lib/receptor_controller/client/response_worker.rb

```diff
@@ -20,8 +20,8 @@ def start
         return if started.value
 
         started.value         = true
-        workers[:maintenance] = Thread.new { check_timeouts }
-        workers[:listener]    = Thread.new { listen }
+        workers[:maintenance] = Thread.new { check_timeouts while started.value }
```

Comment:
Looking at the 2 referenced Satellite PRs - why do we need the `while started.value` conditional that wasn't included in either of those PRs?

Reply:
I agree, inside that thread there is another `while started.value`, not sure why we would need double checks on there. Unless that's just to keep it going in the case of an exception, is that it?

Reply:
yeah, it seems this line is out of date